modules/standard/Heap.chpl

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright 2004-2020 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* 
+  This module contains the implementation of the heap type.
+
+  A heap is a specialized tree-based data structure
+  that supports extracting the maximal or the minimal element quickly,
+  which can serve as a priority queue.
+
+  * Both `push` and `pop` operations are O(lgN).
+  * Querying the top element is O(1).
+  * Initialization from an array is O(N).
+*/
+module Heap {
+  import ChapelLocks;
+  private use HaltWrappers;
+  private use List;
+  private use IO;
+
+  public use Sort only defaultComparator, DefaultComparator,
+                       reverseComparator, ReverseComparator;
```

Comment:
Of these four symbols, I see only `defaultComparator` in the module.  Do you need to include the others?

Reply:
Yes. Only one is used. Just for convenience and I don't have strong opinions. But it's worth mention that it's the same with  Search module: They are all imported but only one is used.